thirdeye/thirdeye-frontend/app/pods/components/rootcause-legend/component.js

```diff
@@ -0,0 +1,40 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  entities: null, // {}
+
+  selectedUrns: null, // Set
+
+  invisibleUrns: null, // Set
+
+  onVisibility: null, // function (Set, bool)
+
+  onSelection: null, // function (Set, bool)
+
+  labels: Ember.computed(
+    'entities',
+    'selectedUrns',
+    function () {
+      const { entities, selectedUrns } = this.getProperties('entities', 'selectedUrns');
+      const labels = {};
+      [...selectedUrns].filter(urn => urn in entities).forEach(urn => labels[urn] = entities[urn].label);
+      return labels;
+    }
+  ),
+
+  actions: {
+    toggleVisibility(urn) {
+      const { onVisibility, invisibleUrns } = this.getProperties('onVisibility', 'invisibleUrns');
+      if (onVisibility) {
+        const state = invisibleUrns.has(urn);
+        onVisibility(urn, state);
+      }
+    },
+    removeUrn(urn) {
+      const { onSelection } = this.getProperties('onSelection');
```

Comment:
Non-blocking: since you're only getting 1 property, you can just use `this.get('onSelection')`

Reply:
I'm inconsistently consistent :)